ddtrace/span.py

```diff
@@ -283,9 +284,19 @@ def set_tag(self, key, value=None):
 
         elif key == MANUAL_KEEP_KEY:
             self.context.sampling_priority = USER_KEEP
+            self.context._update_upstream_services(
```

Comment:
Should this code just be added to the getter/setter for sampling_priority? What if someone sets context.sampling_priority manually instead of using set_tag?

Reply:
Unfortunately we cannot since we need to have the Span (it's service name) when updating this entry.